src/tui.rs

```diff
@@ -364,6 +364,54 @@ impl Tui {
 		}
 	}
 
+	/// Map a terminal cell click (col, row) to a displayed page and device pixel coords.
+	/// Returns (page_index, device_x_px, device_y_px) if the click hits a page image.
+	pub fn map_click_to_page(&self, col: u16, row: u16, full_layout: &RenderLayout, font_size: FontSize) -> Option<(usize, f32, f32)> {
+		let mut img_area = full_layout.page_area;
+
+		// Determine which pages are currently being displayed by reconstructing the same logic
+		// used in `render` to compute `to_display` positions.
+
+		let mut page_widths = self.rendered[self.page..]
+			.iter()
+			// only take those that have images ready
+			.take_while(|p| p.img.is_some())
+			.map(|p| p.img.as_ref().map(|img| (img.w_h().0, img)))
+			.filter_map(|o| o)
```

Comment:
```suggestion
			.filter_map(|p| p.img.as_ref().map(|img| (img.w_h().0, img)))
```

Or maybe you could do


```suggestion
			.filter_map(|p| p.img.as_ref())
			.map(|o| (img.w_h().0, img))
```

But I can't remember if the borrow checker will allow that